### DISCUSSION TITLE
I have a question about the codec tags

### DISCUSSION THREAD

**Discussion by joelRVC** [ggarra13/mrv2#308] [Q&A] [answered]
I have a question about the codec tags with which the videos I save in mrv2 are exported. When I want to export a video from an EXR sequence and play it in an external player (in this case VLC) the contrast is darker. I figured it's because of the transfer function tag, but when I try to change to another option in the export properties the video is always written with sRGB as the transfer function, which causes some players to reproduce the color incorrectly. My question is, how should I configure the parameters to change that tag?

example:

mrv2 view

![{9CFBDFE2-EFC0-4D6E-BA79-B0A2791A03C7}](https://github.com/user-attachments/assets/90b5965b-84ba-45cf-972d-dfc268f8903d)

![{EFE78A13-712F-411C-BD89-9647B83DE9A9}](https://github.com/user-attachments/assets/96967bb0-e086-4b81-b740-64e6e41ca8a5)

**Comment by ggarra13** [comment]
Does the video look good when loaded in mrv2?  If it does, VLC and your other players are probably not paying attention to the codec tags, which is obviously wrong.  Probably messing with them won't have any effect.

If mrv2 reproduces it wrong too, then you found a bug.

To answer your question, which I do **not** recommend you change it, you should go to the "Advanced Settings", click on "Override" (important) and change each one of the tags (you **must** set all of them properly).  Note that I found an issue with the advanced Color Space pulldown which says "sRGB" as default when in fact it is "BGR".  To set it to sRGB, you should set it to "bt601".

**Comment by joelRVC** [comment]
> Does the video look good when loaded in mrv2? If it does, VLC and your other players are probably not paying attention to the codec tags, which is obviously wrong. Probably messing with them won't have any effect.

As I tried more settings I realized something, so the answer here would be yes and no. Apparently I was selecting the pc option in the color range and that was the reason why the video in VLC was dark, when exporting with the tv range there were no problems in vlc. I had not noticed that before in mrv2 because in the configuration I had set the video level to from file, returning it to default the video will make it look dark there too.

![{70BAD736-1502-47A9-A23B-ADE8A5C46FAB}](https://github.com/user-attachments/assets/735d644f-583a-46d4-afdf-3df817813ce9)

So yes you were right, changing the codec settings has no effect. But now I have one more doubt, when I choose gamma22 in the transfer function options (with override activated of course) the tag in the video metadata appears as sRGB, but I see that there is another function that is sRGB IEC 61966-2-1 which also writes sRGB as transfer function when exporting, is that expected?

> Note that I found an issue with the advanced Color Space pulldown which says "sRGB" as default when in fact it is "BGR". To set it to sRGB, you should set it to "bt601".

as far as I understand bt.601 uses slightly different primaries than sRGB, in fact bt.709 would be the correct choice to use as color space and primaries (it uses the same primaries as sRGB), the difference would be mainly in the transfer function, which it would be left in srgb. I think it would be better if those were the default values.

**Reply by ggarra13** [reply]
> as far as I understand bt.601 uses slightly different primaries than sRGB, in fact bt.709 would be the correct choice to use as color space and primaries (it uses the same primaries as sRGB), the difference would be mainly in the transfer function, which it would be left in srgb. I think it would be better if those were the default values.

You are indeed correct, of course.  Those are what are used when Override is off.
I should probably update the defaults to those in the Advance Settings panel so you start with good settings (I currently I just pick the first item in the list).

**Comment by joelRVC** [comment]
I'm still curious about what it means to set the color range on TV or PC? what's going on under the hood

**Reply by ggarra13** [reply]
> I'm still curious about what it means to set the color range on TV or PC? what's going on under the hood

TV limits the colors between the legal range of 16...240 rgb (it is due to how old NTSC/PAL tv sets worked).  PC uses the full range (0..255).

**Reply by joelRVC** [reply]
Curious, because so far with the tests I did it is with the TV option that I get the correct values ​​similar to what I see in the program. As I mentioned before, when using PC and having the video levels at full range, the video becomes too dark, as if I were doing the range calculation twice. Only with the video level in "from file" in the case of PC (using TV chossing "from file" or "full range" shows the same result) do I get the look of the original sequence.

**Reply by ggarra13** [reply]
It is a bug.  The preference setting has the menu items in the wrong order.  Good catch!

**Reply by ggarra13** [reply]
Please try these betas of v1.2.8 that should fix the wrong Preference setting:

.exe:
https://mega.nz/file/mHIW0aAD#EBPnsRqRdS1ExVrF_Zb121YPMSkadQCaA9DhyZbcNYc

.zip:
https://mega.nz/file/eXpWjYgI#Cc8RCrQnwP6Cn5OX_MBiXf0q_vnbP8clhKfilFi6_sE

If you still see wrong behavior between Full Range and Legal Range, please open an Issue with an EXR image, the movie codec and settings you use to save it, and the preferences or menu options you are using to view it.  If you can, also provide screen captures.

**Reply by joelRVC** [reply]
What should be the expected behavior? to confirm correctly

**Reply by ggarra13** [reply]
Full Range and From FIle should show the same color if you have saved the video as "PC (Full Range)".  Full Range should match pretty closely the original EXR.  
Legal Range should show the same color if you have saved the video as "TV (Legal Range)", albeit it may not show exact colors as the EXR (colors may be crushed).
If you set the Preference Settings, the setting should match what is displayed in the Render->Video Levels menu once you exit the Preferences.
If you save as PC, and display as Legal, or viceversa, I am not quite sure how they would display.  I think PC with Legal may also look crushed, while TV with Full may look bright.

**Reply by joelRVC** [reply, marked as answer]
Oh I understand, so I confirm that now the behavior is just as you describe. 

I think it is a good idea if you modify the name of the ranges in the export options as you described them, example: 
- PC --> PC(Full range) , TV -->TV (Legal Range)
or you can make tooltips appear on the buttons when passing over them.
